apps/web/modules/bookings/components/BookingsCalendar.tsx

```diff
@@ -1,23 +1,76 @@
 "use client";
 
 import type { Table as ReactTable } from "@tanstack/react-table";
+import { useMemo, useCallback } from "react";
 
-import { DataTableFilters, DataTableSegment } from "@calcom/features/data-table";
-import { EmptyScreen } from "@calcom/ui/components/empty-screen";
+import {
+  DataTableFilters,
+  DataTableSegment,
+  useDataTable,
+  useFilterValue,
+  ZDateRangeFilterValue,
+  ColumnFilterType,
+} from "@calcom/features/data-table";
+import { CUSTOM_PRESET } from "@calcom/features/data-table/lib/dateRange";
 
 import type { RowData, BookingListingStatus } from "../types";
+import { WeekCalendarView } from "./WeekCalendarView";
 
 type BookingsCalendarViewProps = {
   status: BookingListingStatus;
   table: ReactTable<RowData>;
 };
 
+const COLUMN_IDS_TO_HIDE = ["dateRange"];
+
 export function BookingsCalendar({ table }: BookingsCalendarViewProps) {
+  const { rows } = table.getRowModel();
+  const { updateFilter } = useDataTable();
+  const dateRange = useFilterValue("dateRange", ZDateRangeFilterValue)?.data;
+
+  const bookings = useMemo(() => {
+    return rows
+      .filter((row) => row.original.type === "data")
+      .map((row) => (row.original.type === "data" ? row.original.booking : null))
+      .filter((booking): booking is NonNullable<typeof booking> => booking !== null);
+  }, [rows]);
+
+  const handleWeekChange = useCallback(
+    (startDate: Date, endDate: Date) => {
```

Comment:
If the current week exceeds the applied date range filter, it updates the filter value.